ws-android/Dockerfile

```diff
@@ -5,65 +5,180 @@ FROM cloud9/workspace
 # https://github.com/lukin0110/docker-android-build
 MAINTAINER Jeremy Ellis <keyfreemusic@gmail.com>
 
+# -------------------------- Startup Stuff -----------------------------------------
+# move files from the files folder to the docker container
 ADD ./files/workspace /home/ubuntu/workspace
 
-RUN chmod -R g+w /home/ubuntu/workspace && \
+# Give owner and ubuntu group: read,write,execute permissions, others read only
+# Change owner to ubuntu
+RUN chmod -R 774 /home/ubuntu/workspace && \
     chown -R ubuntu:ubuntu /home/ubuntu/workspace
     
+    
+
+# Setup a bash file to auto load environment variables when in cloud 9
+# Add to this file for every environment variable and Path used
+RUN touch /etc/profile.d/android.sh && \
+    chmod -R 775 /etc/profile.d/android.sh && \
+    chown -R ubuntu:ubuntu /etc/profile.d/android.sh && \
+    printf "#!/bin/bash\n\n" > /etc/profile.d/android.sh  && \
+    printf "echo 'Setting up environment variables'\n" >>               /etc/profile.d/android.sh    && \
+    printf "JAVA_HOME /usr/lib/jvm/java-7-openjdk-amd64\n" >>           /etc/profile.d/android.sh    && \
+    printf "PATH \$PATH:\$JAVA_HOME/bin\n\n" >>                         /etc/profile.d/android.sh    
+
+
+# These commands just let us test the enviornment before submitting to cloud 9    
+ENV JAVA_HOME /usr/lib/jvm/java-7-openjdk-amd64
+ENV PATH $PATH:$JAVA_HOME/bin
+ 
+ 
+ 
+ 
+ 
+
+# -------------------------- Install Necessary packages ----------------------------------------- 
 # Update, upgrade and install packages
 RUN \
     apt-get update && \
     apt-get -y install curl unzip python-software-properties software-properties-common lib32stdc++6 lib32z1
+    
+    
+
 
+
+
+# -------------------------- Install the Android SDK -----------------------------------------
+
+# refer to https://developer.android.com/sdk/index.html#Other
 # Install Android SDK
-# https://developer.android.com/sdk/index.html#Other
 RUN SDK_VERSION=r24.3.3 && \
     set -x && \
     cd /usr/local/ && \
     curl -L -o android-sdk.tgz http://dl.google.com/android/android-sdk_${SDK_VERSION}-linux.tgz && \
     tar xf android-sdk.tgz && \
     rm android-sdk.tgz && \
-    chmod -R +r /usr/local/android-sdk-* && \
-    bash -c 'for file in $(find /usr/local/android-sdk-*); do [[ -x $file ]] && chmod +x "$file"; done'; \
-    exit 0
+    chmod -R 774 /usr/local/android-sdk-linux && \
```

Comment:
Same permissions problem as on /home; the bash line being removed above does what you wanted here, just change the `-x` to `-d`, and the `chmod -R  +r` to `chmod -R +rw`